Allow File::Temp handles to be guessed correctly

waterkip opened the pull request:
The previous implementation only allowed URI objects to be guessed
correctly. This changeset implements $file->can('') so we can deal with
other kind of objects as well. If an object cannot `path()` or
`filename()` we stringify the object as a last resort.

oalders commented:
I think this would be easier to review without the whitespace and tidying changes.  Ideally those would be in a different PR.

oalders commented:
Oh, I see the tidying is in different commits.  That's easier to review.  👍

oalders commented on lib/LWP/MediaTypes.pm:
```diff
@@ -40,61 +40,68 @@ my %suffixEncoding = (
+            $fullname = $file->filename;
+        }
+        else {
+            $fullname = "" . $file;
```
This condition doesn't appear to get exercised by the tests.  It would be helpful to get some test coverage for this new code.

oalders replied:
Also, if we merge this, we'll need to update the documentation to explain the new behaviour.

waterkip replied:
I've added two fixup commits for this. If you agree with the documentation changes I'll autosquash them and push them again. Thanks for the feedback.

oalders committed:
Add Travis CI config file

waterkip committed:
Allow File::Temp handles to be guessed correctly

The previous implementation only allowed URI objects to be guessed
correctly. This changeset implements `$file->can()` so we can deal
with other kind of objects as well. If an object cannot `path()` or
`filename()` we stringify the object as a last resort.

Signed-off-by: Wesley Schwengle <wesley@schwengle.net>
